### PR TITLE
VmList: Add Create VM Wizard

### PR DESCRIPTION
**Pull request by mareklibra**
Depends on:
- [x] #1821 
- [x] #1820
- [x] #1819 
- [x] #1828 - kubevirt-web-ui-components release

### CODE REVIEW

File: frontend/packages/console-shared/src/modals/modal-resource-launcher.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        {message}
+      </div>
+    )}
+  </Box>
+);
+
+const ModalComponentWrapper = (props: ComponentWrapperProps) => {
+  const { Component, closeModal, resourceToProps, loadError, loaded, ...restProps } = props;
+
+  if (loadError) {
```
Comment (suomiy) [inline frontend/packages/console-shared/src/modals/modal-resource-launcher.tsx]: shouldn't we rather check for `!loaded` here?

File: frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  ];
+
+  const resourcesToProps = (props) => {
+    const { namespaces, userTemplates, commonTemplates } = props;
+    const flatten = (resourceName) => ({ [resourceName]: _.get(props[resourceName], 'data') });
+
+    let selectedNamespace;
+    if (activeNamespace) {
+      selectedNamespace = _.get(namespaces, 'data', []).find(
+        (namespace) => namespace.metadata.name === activeNamespace,
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx]: we can use selector here

File: frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      );
+    }
+
+    let templates;
+    if (userTemplates && commonTemplates) {
+      templates = _.get(userTemplates, 'data', []).concat(_.get(commonTemplates, 'data', []));
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx]: can we use spread operator here instead?

File: frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    // namespaces, templates, dataVolumes, virtualMachines
+    return {
+      selectedNamespace,
+      templates,
+
+      ...flatten('namespaces'),
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx]: can we use map here? or reduce..

File: frontend/packages/console-shared/src/modals/modal-resource-launcher.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import * as ReactDOM from 'react-dom';
+import * as _ from 'lodash';
+import { Provider } from 'react-redux';
+import { Router } from 'react-router-dom';
+
+import store from '@console/internal/redux';
+import { Firehose, AccessDenied, history, MsgBox, Box } from '@console/internal/components/utils';
+import { ModalErrorContent } from '@console/internal/components/modals/error-modal';
+
+const NotFound = ({ message }) => (
```
[the remark this answers is not in the document]
Comment (suomiy) [reply frontend/packages/console-shared/src/modals/modal-resource-launcher.tsx]: shouldn't this rather be in a separate file or in `@console/internal/components/utils` ?

File: frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx (hunk 4 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx]: I think recent syntax is much more readable. The list of names would still need to be somewhere so I don't see much benefit it changing that unless there is other strong opinion on that.

File: frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx (hunk 2 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx]: done

**Comment by mareklibra** [comment]
Rebased

File: frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx (hunk 4 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx]: ok, it is not a must

**Comment by mareklibra** [comment]
Rebased

**Comment by mareklibra** [comment]
Rebased

**Comment by mareklibra** [comment]
Rebased as all dependencies are merged now.

**Comment by jelkosz** [comment]
already reviewed by Filip, adding my formal
/lgtm

**Pull request merged**